### ISSUE TITLE
Fraction absorbed not calculated properly

### ISSUE THREAD

**Issue by msevestre** [Open-Systems-Pharmacology/PK-Sim#1198]
This is not #968

**Comment by Yuri05** [comment]
That's why it's fraction absorbed **_into mucosa_** (and not into the portal vein).
s. also #517

**Comment by msevestre** [comment]
No idea what you are talking about
![image](https://user-images.githubusercontent.com/1041237/56620551-52ae6880-65f7-11e9-8dbc-72c3fb728eba.png)

It says Fraction absorbed. User is expecting something else. For me it's a clear bug

**Comment by msevestre** [comment]
Could not we solve this with some clever use of SumFormula BTW?

**Comment by Yuri05** [comment]
Yep, this one is fraction absorbed into mucosa

**Comment by msevestre** [comment]
Do you agree that it's not written anywhere? Either we calculate fraction absorbed (everyone assumes portal veine) or we rename this parameter

**Comment by msevestre** [comment]
`Fraction absorbed` is a pharmacokinetic parameter that has a very specific meaning.

**Comment by sfrechen** [comment]
No, fraction absorbed is really the fraction of the dose that is absorbed into the gut wall mucoca (enterocytes), usually abbreviated with Fa. The fraction escaping gut wall metabolism and reaching the portal vein is Fg. Finally, we have the fraction escaping first pass liver metabolism, which is called Fh. The product Fa*Fg*Fh is the absolute bioavailability F after oral dosing.

Midazolam, for example, comes with an F of around 25%. In this specific case, Fg and Fh are both approx. 50%, while still everything is absorbed and thus an Fa of 1 would be correct!

**Comment by StephanSchaller** [comment]
![image](https://user-images.githubusercontent.com/25477643/56638724-30583180-666f-11e9-99aa-4a32c4844f59.png)

**Comment by StephanSchaller** [comment]
![image](https://user-images.githubusercontent.com/25477643/56639027-f5a2c900-666f-11e9-9f6d-219fe6b4640e.png)

Calculated, it is always "amount of outflow divided by amount of inflow"

- fa would be the sum of the integral over time of all absorption rates from gut lumen to mucosa divided by dose
- fG would be the sum of the integral over time of all absorption rates from mucosa to portal vein divided by the sum of the integral over time of all absorption rates from gut lumen
- simplified fH would be liver outflow divided by liver inflow

**Comment by sfrechen** [comment]
Having Fg calculated directly by PK-Sim would be a nice feature.
(EHC may always give abstruse results as for Fa as well.)

**Comment by PavelBal** [comment]
That image of @StephanSchaller should be in the wiki, if it is still not.

**Comment by Yuri05** [comment]
Ok, I changed the issue title into _Fraction absorbed should be renamed to "Fabs into mucosa"_
What is calculated now is calculated correctly - it might be just rename for the sake of clarity.

**Comment by msevestre** [comment]
>What is calculated now is calculated correctly - it might be just rename for the sake of clarity

Lol. "We calculate something the right way, we just call something different that is calculated another way"

**Comment by msevestre** [comment]
The image above is great indeed and should be in the documentation!
Do you guys want to make a PR?

**Comment by StephanSchaller** [comment]
> Do you guys want to make a PR?

What is that? :-)

**Comment by msevestre** [comment]
Pull Request: https://docs.open-systems-pharmacology.org/how-to-contribute

Pronounced "Pee Arr".  :)

**Comment by StephanSchaller** [comment]
☠️ Arrr!

**Comment by msevestre** [comment]
@sfrechen @Yuri05  
>Having Fg calculated directly by PK-Sim would be a nice feature.

There is a proposal for the calculation in #517 
however I am wondering why we would not be able to calculate it directly using observer. Am I missing something?

**Comment by StephanSchaller** [comment]
> however I am wondering why we would not be able to calculate it directly using observer. Am I missing something?

I agree, as mentioned above a parameter/observer with sum formula should do the trick.

> fG would be the sum of the integral over time of all absorption rates from mucosa to portal vein divided by the sum of the integral over time of all absorption rates from gut lumen

If this feature should also be available for the protein model, we just have to consider lymph flow as an escape route ;-)

**Comment by Yuri05** [comment]
@msevestre @StephanSchaller 
Without doing something like proposed in #517 it's not possible to determine which proportion of the flow `mucosa=>portal vein` comes from absorption and which from the system.
![grafik](https://user-images.githubusercontent.com/25061876/56692183-e78d9080-66e1-11e9-96de-ac8c5e184a74.png)

**Comment by StephanSchaller** [comment]
> it's not possible to determine which proportion of the flow mucosa=>portal vein comes from absorption and which from the system.

Why not?
You can access each single rate of flow/diffusion/transport. Just add and substract them accordingly…?

Sum up all rates of para-cellular absorption = flow_absorbed_para
Sum up all rates of trans-cellular absorption = flow_absorbed_trans
Sum up all rates of mucosal cell to interstitial diffusion/transport = f_g_trans

f_g = IntegralOverTime((flow_absorbed_para + f_g_trans)/(flow_absorbed_para + flow_absorbed_trans))

Well, it might not be so easy to define flow_absorbed_trans and f_g_trans if user-specific transporters are defined… but should be possible to find a way to automatically include these as well?

**Comment by Yuri05** [comment]
Transporters are not the problem. The problem is the systemis flow on the right side of the picture above, which you cannot separate from the absorption.

>Sum up all rates of para-cellular absorption = flow_absorbed_para

easy (in fact, already done)

>Sum up all rates of trans-cellular absorption = flow_absorbed_trans

easy (in fact, already done)

>Sum up all rates of mucosal cell to interstitial diffusion/transport = f_g_trans

Will also include the "systemic" part and the result will be wrong.

In order to calculate Fracton absorbed into portal vein we must
a) Set all the systemic flows to mucosa = 0
b) integrate over the `sum of all flows Mucosa=>portal vein` and divide it by the dose

Or alternatively do like proposed in #517 (which will de-facto also deactivate systemic flows to mucosa)

If there would be no metabolism in mucosa and (at the same time) in case of absorption sink conditions, it would be possible to calculate Fabs_PortalVein by integrationg over `Sum_of_Flows_Mucosa_PortalVein - Sum_of_flows_System_to_Mucosa`. And here an observer would do. In other cases not, and the only general solution I see is by deactivating systemic flows to mucosa

**Comment by msevestre** [comment]
Stupid question but: Where is this picture coming from? What is "Submucosa + Muscularis + Serosa" ?

Isn't it possible to substract the systemic part ? (e.g. rate flowing into mucosa that  are not part of absorption?)

**Comment by StephanSchaller** [comment]
> Will also include the "systemic" part and the result will be wrong.

Well, that's a matter of perspective, I guess ;-)
Technically, the systemic flow also distorts trans- & para-cellular absorption as both are bi-directional.

Of course you will have a "systemic part" in your evaluation of F_oral = fa*Fg*Fh … you always do. That's why F_oral can be > 1.

And (technically) that's okay, because if systemic drug reaches cell mucosa to be metabolized (thereby reducing "f_g_trans", it is not escaping gut metabolism.
And you have to simulate to the "end" and calculate the IntegralOverTime for flows to "balance out".
F_g is really just about that flow balance:

> IntegralOverTime((flow_absorbed_para + f_g_trans)/(flow_absorbed_para + flow_absorbed_trans))

no matter "where (& when) the drug comes from".

**Comment by StephanSchaller** [comment]
Technically, #517 (only) calculates the "First-pass Fg"

**Comment by Yuri05** [comment]
>Stupid question but: Where is this picture coming from? What is "Submucosa + Muscularis + Serosa" ?

https://www.fda.gov/downloads/Drugs/NewsEvents/UCM503767.pdf

>Isn't it possible to substract the systemic part ? (e.g. rate flowing into mucosa that are not part of absorption?)

It's possible, but only in case there is no metabolism in mucosa and sink conditions.

>Technically, the systemic flow also distorts trans- & para-cellular absorption as both are bi-directional.

correct (in case there are NO absorption sink conditions and/or active transports mucosa=>lumen)

>Of course you will have a "systemic part" in your evaluation of F_oral = faFgFh … you always do. That's why F_oral can be > 1.

Which does not make any sense for me. But as you said: that's a matter of perspective :eyes:

>And (technically) that's okay, because if systemic drug reaches cell mucosa to be metabolized (thereby reducing "f_g_trans", it is not escaping gut metabolism.
And you have to simulate to the "end" and calculate the IntegralOverTime for flows to "balance out".
F_g is really just about that flow balance: no matter "where (& when) the drug comes from".

Well, the only calculation which makes sense in my eyes is by deactivating the systemic flow.
Everything else is garbage. Just IMHO.

**Comment by msevestre** [comment]
>It's possible, but only in case there is no metabolism in mucosa and sink conditions.

Why? Metabolism rates are accounted for separately. They are not part of the absorption. 

>Submucosa + Muscularis + Serosa

Do we have that in our models?

Sink Condition maybe

**Comment by Yuri05** [comment]
>Why? Metabolism rates are accounted for separately. They are not part of the absorption.

?? we are talking about fraction absorbed into portal vein, right? Thus metabolism in mucosa must be also taken into account.

>Do we have that in our models?

Yes (as "all-in-one", not subdivided into those parts)

**Comment by msevestre** [comment]
I am talking about fg (the fraction escaping metabolism).  
I would NOT call that fraction absorbed!

Let's talk about it. I just don't understand what's the issue with metabolism processses. 

If there is a way to calculate it on the FLY instead of doing another simulation, we should do it

Also since F = fa*FG * Fh, maybe only calculating the first pass Fg is actually incorrect? (because F>1)

**Comment by StephanSchaller** [comment]
> Everything else is garbage. Just IMHO.

Well...let's look at F_oral > 1. That happens once you have EHC and the drug get's re-absorbed after that.
And this is what users expect.
Meaning, you actually should NOT "remove" the "systemic part" in your evaluation of F_oral.

But, if you just want the non-systemic-influenced "first-pass" F_oral. You will need to do it with the sink-condition simulation.
And as you are doing this already anyway, it is not a big issue. But as mentioned in #517 you will need to divide the PV fractional amount with fa to get Fg.

**Comment by sfrechen** [comment]
Regarding F_oral: we need to stick to the definition! That is F_oral = (AUCinf_po/Dose_oral)/(AUC_inf_iv/Dose_iv).
Thus, we need to account for EHC and so on. EHC also happens after iv administration.

**Comment by StephanSchaller** [comment]
> Regarding F_oral: we need to stick to the definition! That is F_oral = (AUCinf_po/Dose_oral)/(AUC_inf_iv/Dose_iv).

True. This also cancels "systemic influence" out of the equation (except maybe if the PO and IV reference doses differ by much and you have nonlinearities).
So if we stick to the definition (and I guess we should do that) we will either have to simulate the reference scenarios or the "truncated" model with sink conditions.

**Comment by AndreDlm** [comment]
Yes, using the true definition for F_oral is important. And Stephan's proposal to estimate a "first pass Fg" would be an adequate solution in my opinion
